consensus/src/state_computer.rs

```diff
@@ -194,6 +200,7 @@ impl StateComputer for ExecutionProxy {
 
     /// Synchronize to a commit that not present locally.
     async fn sync_to(&self, target: LedgerInfoWithSignatures) -> Result<(), StateSyncError> {
+        let _guard = self.write_mutex.lock().await;
```

Comment:
Just to double check: is this sequence of events possible?
1. We commit some blocks (using the write_mutex) but there's still more pending.
2. sync_to is called and the write mutex is held while state sync gets us to the target.
3. sync_to drops the write mutex and then commit immediately grabs it again to commit the blocks pending in 1?

If this happens, will we commit invalid data? e.g., step 1 takes us to version `V`, step 2 takes us to `Target > V` and step 3 tries to commit data between `V` and `Target`?

I don't know the code, so just double checking 😄

Reply:
yeah the change to `shutdown_current_processor` is because of exact issue you described 😄 I think we now purge all the commit queue before calling sync_to.